doc/develop/acq.rst

```diff
@@ -372,3 +372,70 @@ SPARC alignment:
 In mirror alignment mode, during background acquisition, the SEM is stopped to
 automatically activate the blanker (which will happen only if it's in the automatic setting).
 
+Optical Path Manager
+====================
+The goal of the Optical Path Manager is to the position actuators affecting the
+light path such that they are in the right configuration for a given acquisition type.
+It is found in ``acq.path.OpticalPathManager`` and is typically used as a singleton.
+The GUI takes care of instantiating it at init, and then it can be accessed via
+``MainGUIData.opm``. In practice, this is almost exclusively used on the SPARC.
+
+
+It has one important method:
+
+.. py:method:: setPath(mode, detector=None)
+
+   Sets all the necessary components of the optical path to the positions
+   required to achieve the requested mode.
+
+   :param path: The stream or the optical path mode
+   :type path: stream.Stream or str
+   :param detector:
+      When the `path` is a str, this allows
+      to define which detector will be targeted on this path. That
+      is useful in case the mode can be used with multiple detectors (eg,
+      ``fiber-align`` on a SPARC with multiple spectrometers). When path is a
+      Stream, the ``Stream.detector`` is always used.
+   :type detector: Component or None
+   :return: a Future allowing to follow the status of the path update.
+   :rtype: Future
+   :raise: ValueError if the given mode does not exist
+           IOError if a detector is missing
+
+The list of supported `paths` (ie, optical modes) can be found in the ``acq.path`` module.
+At runtime, they are limited by the microscope type and the available components.
+
+In terms of implementation, for each optical mode, there are two types of actuators:
+
+ * The path selectors which, typically, are mirrors with 2 positions. Their positions
+   is derived from the "affects" property of each component, by computing a graph
+   joining the actuators to the detectors.
+ * The acquisition type actuators which configure how the light can be seen. For
+   instance in AR mode, the lens 2 needs to be activated, while it should be removed
+   during spectrum mode.
+
+Currently, in Odemis, the Optical Path Manager is called from two places:
+
+ * In the ``Stream.prepare()``, which is itself called just before a stream acquires
```

Comment:
I don't understand this sentence, especially "which is itself".

Reply:
I mean that "Stream.prepare()" calls the optical path manager. Then, the reader could immediately ask "when is stream.prepare() called?". It's called just before a stream acquisition starts.
Do you understand what I mean? How would you express it better?
How about:
"In ``Stream.prepare()``. This function is in charge of preparing all the
   microscope components prior to the stream acquisition.
   In Odemis, it is called explicitly in some cases. It is also called implicitly,
   every time a stream is about to start acquiring data."
Is that clearer?

Reply:
That's more clear.